src/libraries/Native/Unix/System.Native/pal_io.c

```diff
@@ -1405,11 +1409,31 @@ int64_t SystemNative_GetFileSystemType(intptr_t fd)
     // which got deprecated in macOS 10.6, in favor of statfs
     while ((statfsRes = fstatfs(ToFileDescriptor(fd), &statfsArgs)) == -1 && errno == EINTR) ;
     return statfsRes == -1 ? (int64_t)-1 : (int64_t)statfsArgs.f_type;
+#elif defined(TARGET_SUNOS // we use statvfs (see SystemNative_GetFileSystemTypeAsString) on SunOS
```

Comment:
```suggestion
#elif defined(TARGET_SUNOS) // we use statvfs (see SystemNative_GetFileSystemTypeAsString) on SunOS
```

Reply:
Would it be better to not add the additional export and instead just do a translation here from file system name to expected ID?

Reply:
It would result in duplicating the enum entries that we have on the managed side.

The `f_type` in `struct statfs` is not available with the macros we have defined. I read the code and it seems like even if we change the macro definitions for system header to reveal this field, we will still get zero: https://github.com/illumos/illumos-gate/blob/9ecd05bdc59e4a1091c51ce68cce2028d5ba6fd1/usr/src/ucbhead/sys/vfs.h#L241
cc @jclulow, @sjorge in case I've missed anything :)